### PR TITLE
Add CDP4Reporting

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Add CDP4Reporting

### CODE REVIEW

File: CDP4Reporting/CDP4Reporting.csproj [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+	  <TargetFrameworks>netstandard2.0</TargetFrameworks>
+	  <Company>RHEA System S.A.</Company>
+	  <Title>CDP4Reporting Community Edition</Title>
+	  <VersionPrefix>1.0.0</VersionPrefix>
+	  <Description>CDP4 Reporting</Description>
+	  <Copyright>Copyright © RHEA System S.A.</Copyright>
+	  <Authors>Sam, Merlin, Alex, Naron, Alexander, Yevhen, Nathanael</Authors>
```
Comment (samatrhea) [inline CDP4Reporting/CDP4Reporting.csproj:10]: Please only keep Alex, Sam and Alexander for now

File: CDP4Reporting/CDP4Reporting.csproj [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	  <PackageId>CDP4Reporting-CE</PackageId>
+	  <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+	  <PackageProjectUrl>http://sdk.cdp4.org</PackageProjectUrl>
+	  <PublishRepositoryUrl>true</PublishRepositoryUrl>
+	  <PackageIcon>cdp4-icon.png</PackageIcon>
+	  <IncludeSource>true</IncludeSource>
+	  <IncludeSymbols>true</IncludeSymbols>
+	  <RepositoryUrl>https://github.com/RHEAGROUP/COMET-SDK-Community-Edition.git</RepositoryUrl>
+	  <PackageTags>CDP COMET ECSS-E-TM-10-25 JSON</PackageTags>
```
Comment (samatrhea) [inline CDP4Reporting/CDP4Reporting.csproj:19]: remove JSON

File: CDP4Reporting/DataCollection/DataCollectorCategory.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="DataCollectorCategory.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2022 RHEA System S.A.
+//
+//    Author: Sam Gerené, Merlin Bieze, Alex Vorobiev, Naron Phou, Alexander van Delft, Nathanael Smiechowski
```
Comment (samatrhea) [inline CDP4Reporting/DataCollection/DataCollectorCategory.cs:5]: please remove all authors on all files and keep Alex, Sam and Alexander for now

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4Reporting.Tests/CDP4Reporting.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,70 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net48</TargetFramework>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
+    <AssemblyTitle>CDP4BuiltInRules.Tests</AssemblyTitle>
+    <Company>RHEA System S.A.</Company>
+    <Product>CDP4BuiltInRules.Tests</Product>
```
Comment (samatrhea) [inline CDP4Reporting.Tests/CDP4Reporting.Tests.csproj:7]: please update to the correct name

File: CDP4Reporting.Tests/DataCollection/CategoryDecompositionHierarchyTestFixture.cs [hunk 5]
```diff
@@ -0,0 +1,153 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CategoryHierarchyTestFixture.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2022 RHEA System S.A.
```
Comment (samatrhea) [inline CDP4Reporting.Tests/DataCollection/CategoryDecompositionHierarchyTestFixture.cs]: 2023 (please update all over)

File: CDP4Reporting/DataCollection/DataCollectorBooleanParameter.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    using CDP4Common.Helpers;
+
+    /// <summary>
+    /// Abstract base class from which all bool parameter columns
+    /// for a <see cref="DataCollectorRow"/> need to derive.
+    /// </summary>
+    /// <typeparam name="TRow">
+    /// The type of the associated <see cref="DataCollectorRow"/>.
+    /// </typeparam>
+    public class DataCollectorBooleanParameter<TRow> : DataCollectorParameter<TRow, bool>
```
Comment (samatrhea) [inline CDP4Reporting/DataCollection/DataCollectorBooleanParameter.cs:38]: docs state that this is an abstract base class, while it is not

File: CDP4Reporting/DataCollection/DataCollectorBooleanParameter.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        /// <summary>
+        /// Parses a parameter value as double.
+        /// </summary>
+        /// <param name="value">
+        /// The parameter value to be parsed.
+        /// </param>
+        /// <returns>
+        /// The parsed value.
+        /// </returns>
+        [ExcludeFromCodeCoverage] // Remove attribute when more logic is added to this method. It's only SDK functionality now and that is fully covered.
```
Comment (samatrhea) [inline CDP4Reporting/DataCollection/DataCollectorBooleanParameter.cs:50]: please remove this `ExcludeFromCodeCoverage`

File: CDP4Reporting/DataCollection/DataCollectorDoubleParameter.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    using CDP4Common.Helpers;
+
+    /// <summary>
+    /// Abstract base class from which all double parameter columns
+    /// for a <see cref="DataCollectorRow"/> need to derive.
+    /// </summary>
+    /// <typeparam name="TRow">
+    /// The type of the associated <see cref="DataCollectorRow"/>.
+    /// </typeparam>
+    public class DataCollectorDoubleParameter<TRow> : DataCollectorParameter<TRow, double>
```
Comment (samatrhea) [inline CDP4Reporting/DataCollection/DataCollectorDoubleParameter.cs:38]: docs state that this is an abstract class while it is not

File: CDP4Reporting/DataCollection/DataCollectorStateDependentPerRowDoubleParameter.cs [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        /// <summary>
+        /// Parses a parameter value as double.
+        /// </summary>
+        /// <param name="value">
+        /// The parameter value to be parsed.
+        /// </param>
+        /// <returns>
+        /// The parsed value.
+        /// </returns>
+        [ExcludeFromCodeCoverage] // Remove attribute when more logic is added to this method. It's only SDK functionality now and that is fully covered.
```
Comment (samatrhea) [inline CDP4Reporting/DataCollection/DataCollectorStateDependentPerRowDoubleParameter.cs:50]: please remove `ExcludeFromCodeCoverage`

File: CDP4Reporting/DynamicTableChecker/IDynamicTableCellsCollector.cs [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        /// </summary>
+        /// <param name="tableName">The name of the Table in the report</param>
+        /// <param name="expression">The expression that will be used to display text in the table cell.</param>
+        /// <returns>
+        /// A <see cref="DynamicTableCell"/>
+        /// </returns>
+        DynamicTableCell AddExpressionTableCell(string tableName, string expression);
+    }
+
+    public class DynamicTableCell
```
Comment (samatrhea) [inline CDP4Reporting/DynamicTableChecker/IDynamicTableCellsCollector.cs:66]: please add docs to this class, my preference would also be to put this class in its own file

**Review by samatrhea — CHANGES_REQUESTED** [review]
Not so sure i agree with the use of `ExcludeFromCodeCoverage`, it hides that we don't have coverage... best be honest about it and improve when we can instead of forgetting about it

some other minor comments (also from past review) that need to be fixed

**Review by samatrhea — APPROVED** [review]

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**